testing/can-messages/src/messages.rs

```diff
@@ -952,6 +955,179 @@ impl Into<f32> for DolorOneFloat {
+    pub fn sensor_sonars_err_count_raw(&self) -> u16 {
+        let signal = self.raw.view_bits::<Lsb0>()[4..16].load_le::<u16>();
+
+        signal
+    }
+
+    /// Set value of SENSOR_SONARS_err_count
+    #[inline(always)]
+    pub fn set_sensor_sonars_err_count(&mut self, value: u16) -> Result<(), CanError> {
+        #[cfg(feature = "range_checked")]
+        if value < 0_u16 || 0_u16 < value {
+            return Err(CanError::ParameterOutOfRange { message_id: 200 });
+        }
+        self.raw.view_bits_mut::<Lsb0>()[4..16].store_le(value);
+        Ok(())
+    }
+}
+
+impl core::convert::TryFrom<&[u8]> for SensorSonars {
+    type Error = CanError;
+
+    #[inline(always)]
+    fn try_from(payload: &[u8]) -> Result<Self, Self::Error> {
+        if payload.len() != 8 {
+            return Err(CanError::InvalidPayloadSize);
+        }
+        let mut raw = [0u8; 8];
+        raw.copy_from_slice(&payload[..8]);
+        Ok(Self { raw })
+    }
+}
+
+#[cfg(feature = "debug")]
+impl core::fmt::Debug for SensorSonars {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        if f.alternate() {
+            f.debug_struct("SensorSonars")
+                .field("sensor_sonars_err_count", &self.sensor_sonars_err_count())
+                .finish()
+        } else {
+            f.debug_tuple("SensorSonars").field(&self.raw).finish()
+        }
+    }
+}
+
+#[cfg(feature = "arb")]
+impl<'a> Arbitrary<'a> for SensorSonars {
+    fn arbitrary(u: &mut Unstructured<'a>) -> Result<Self, arbitrary::Error> {
+        let sensor_sonars_mux = u.int_in_range(0..=0)?;
+        let sensor_sonars_err_count = u.int_in_range(0..=0)?;
+        let sensor_sonars_left = u.float_in_range(0_f32..=0_f32)?;
+        let sensor_sonars_middle = u.float_in_range(0_f32..=0_f32)?;
+        let sensor_sonars_right = u.float_in_range(0_f32..=0_f32)?;
+        let sensor_sonars_rear = u.float_in_range(0_f32..=0_f32)?;
+        let sensor_sonars_no_filt_left = u.float_in_range(0_f32..=0_f32)?;
+        let sensor_sonars_no_filt_middle = u.float_in_range(0_f32..=0_f32)?;
+        let sensor_sonars_no_filt_right = u.float_in_range(0_f32..=0_f32)?;
+        let sensor_sonars_no_filt_rear = u.float_in_range(0_f32..=0_f32)?;
+        SensorSonars::new(
+            sensor_sonars_mux,
+            sensor_sonars_err_count,
+            sensor_sonars_left,
+            sensor_sonars_middle,
+            sensor_sonars_right,
+            sensor_sonars_rear,
+            sensor_sonars_no_filt_left,
+            sensor_sonars_no_filt_middle,
+            sensor_sonars_no_filt_right,
+            sensor_sonars_no_filt_rear,
+        )
+        .map_err(|_| arbitrary::Error::IncorrectFormat)
+    }
+}
+/// Defined values for multiplexed signal SENSOR_SONARS
+#[derive(Clone, Copy)]
+#[cfg_attr(feature = "debug", derive(Debug))]
+pub enum SensorSonarsSensorSonarsMux<'a> {
+    SensorSonarsSensorSonarsMuxM0(SensorSonarsSensorSonarsMuxM0<'a>),
```

Comment:
I would be very happy if we could find a shorter/better name for the variants

Reply:
I think for the variants it would probably be sufficient to just name them e.g. M0, M1... for the wrapped structs I think it's probably difficult because of naming collisions.